src/components/layouts/FilterBar.tsx

```diff
@@ -79,85 +79,170 @@ export const FilterBar = ({ updateViewModeSetting }: { updateViewModeSetting: (m
-                store.dispatch(setViewSettings(mode))
-                updateViewModeSetting(mode)
-              }}
-            />
+              <Box
+                sx={{
+                  display: { xs: 'none', sm: 'block' },
+                }}
+              >
+                <ViewModeSelector
+                  selectedMode={view}
+                  handleModeChange={(mode) => {
+                    store.dispatch(setViewSettings(mode))
+                    updateViewModeSetting(mode)
+                  }}
+                />
+              </Box>
+            </Stack>
           </Stack>
+        </AppMargin>
+      </Box>
+      <AppMargin size={SizeofAppMargin.LARGE}>
+        <Stack direction="row" justifyContent="space-between" sx={{ display: { xs: 'flex', sm: 'none' } }}>
+          <Selector
+            getSelectedValue={(_newValue) => {
+              const newValue = _newValue as IAssigneeCombined
+              updateAssigneeValue(newValue)
+              handleFilterOptionsChange(FilterOptions.ASSIGNEE, newValue.id as string)
+            }}
+            startIcon={<FilterByAsigneeIcon />}
+            options={assignee}
+            placeholder="Assignee"
+            value={assigneeValue}
+            selectorType={SelectorType.ASSIGNEE_SELECTOR}
+            extraOption={{
+              id: '',
+              name: 'No assignee',
+              value: '',
+              extraOptionFlag: true,
+            }}
+            extraOptionRenderer={(setAnchorEl, anchorEl, props) => {
+              return (
+                <Box
+                  component="li"
+                  {...props}
+                  sx={{
+                    '&.MuiAutocomplete-option[aria-selected="true"]': {
+                      bgcolor: (theme) => theme.color.gray[100],
+                    },
+                    '&.MuiAutocomplete-option[aria-selected="true"].Mui-focused': {
+                      bgcolor: (theme) => theme.color.gray[100],
+                    },
+                  }}
+                  onClick={(e) => {
+                    updateAssigneeValue({ id: '', name: 'No assignee' })
+                    setAnchorEl(anchorEl ? null : e.currentTarget)
+                    handleFilterOptionsChange(FilterOptions.ASSIGNEE, 'none')
+                  }}
+                >
+                  <Stack direction="row" alignItems="center" columnGap={3}>
+                    <Avatar alt="user" sx={{ width: '20px', height: '20px' }} />
+                    <Typography variant="sm" fontWeight={400}>
+                      No assignee
+                    </Typography>
+                  </Stack>
+                </Box>
+              )
+            }}
+            buttonContent={
+              <FilterByAssigneeBtn
+                assigneeValue={assigneeValue}
+                updateAssigneeValue={updateAssigneeValue}
+                handleClick={handleFilterOptionsChange}
+              />
+            }
+          />
+
```

Comment:
This is really confusing, we have a bit of a prop hell going on here. Is this possible to clean up? Maybe change one prop to children

Reply:
`Selector` component accepts the `buttonContent` prop. `FilterByAssigneeBtn` itself is a reusable component which accepts the above props. I don't think there is a prop hell going on here since we are only drilling prop from parent to child component (level 1 drill).